src/CSET/operators/plot.py

```diff
@@ -141,39 +137,43 @@ def _colorbar_map_levels(
         Variable name to extract from the dictionary
 
     """
+    # Grab the colour bar file from the recipe global metadata. A non-existent
+    # placeholder path is used if not found.
+    colorbar_file = get_recipe_metadata().get(
+        "style_file_path", "/non-existent/NO_FILE_SPECIFIED"
+    )
     try:
         with open(colorbar_file, "rt", encoding="UTF-8") as fp:
             colorbar = json.load(fp)
 
-            # Specify the colormap for this variable
-            try:
-                cmap = colorbar[varname]["cmap"]
-                print("From color_bar dictionary: Using cmap")
-            except KeyError:
-                cmap = mpl.colormaps["viridis"]
+        # Specify the colormap for this variable
+        try:
+            cmap = colorbar[varname]["cmap"]
+            logging.debug("From color_bar dictionary: Using cmap")
+        except KeyError:
+            cmap = mpl.colormaps["viridis"]
 
-            # Specify the colorbar levels for this variable
-            try:
-                levels = colorbar[varname]["levels"]
+        # Specify the colorbar levels for this variable
+        try:
+            levels = colorbar[varname]["levels"]
 
-                actual_cmap = mpl.cm.get_cmap(cmap)
+            actual_cmap = mpl.cm.get_cmap(cmap)
 
-                norm = mpl.colors.BoundaryNorm(levels, ncolors=actual_cmap.N)
-                print("From color_bar dictionary: Using levels")
+            norm = mpl.colors.BoundaryNorm(levels, ncolors=actual_cmap.N)
+            logging.debug("From color_bar dictionary: Using levels")
+        except KeyError:
+            try:
+                vmin, vmax = colorbar[varname]["min"], colorbar[varname]["max"]
+                logging.debug("From color_bar dictionary: Using min and max")
+                levels = np.linspace(vmin, vmax, 10)
+                norm = None
             except KeyError:
-                try:
-                    vmin, vmax = colorbar[varname]["min"], colorbar[varname]["max"]
-                    print("From color_bar dictionary: Using min and max")
-
-                    levels = np.linspace(vmin, vmax, 10)
-                    norm = None
-                except KeyError:
-                    levels = None
-                    norm = None
+                levels = None
+                norm = None
```

Comment:
I unindented this block, as the JSON file didn't need to be kept open after being read. Otherwise the only change is swapping `print()` for `logging.debug()`